doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -39,15 +50,14 @@ Tactics for simple equalities
 
 .. tacn:: reflexivity
 
-   For a goal with the form :n:`{? forall @open_binders , } t = u`,
-   verifies that `t` and `u` are
-   :term:`definitionally equal <definitional equality>`, and if so,
-   solves the goal (by applying `eq_refl`).  If not, it fails.
+   After doing an :tacn:`intros`,
+   if the resulting goal is in the form `t = u` in which `t` and `u` are
```

Comment:
Technically, this is not fully accurate (we can also say that it wasn't fully accurate before). I'm not sure if it is worth fixing though:

```coq
Definition eq_nat (x : nat) := let y := x in x = y.

Goal eq_nat 0.
intros. (* does nothing *)
reflexivity. (* solves the goal *)
```

Reply:
Clearly it does an unfold, a delta reduction.  Strange code.  The tactic code is at `Tactics.intros_reflexivity`,
which calls `reflexivity`,
which calls `reflexivity_red false`,
which calls `maybe_betadeltaiota_concl allowred gl` (allowred is false),
which makes `maybe_betadeltaiota_concl` do nothing.

`intros_reflexivity` then calls `reflexivity`,
which  calls `Rewrite.setoid_reflexivity`,
which eventually calls `reflexivity_red true`--which should do the reduction, but I don't understand the flow of control.

I suppose I could add text saying that the tactic will do beta-delta-iota reduction after the intros.  That would cover your example though I would probably not include your example in the text.

My impression is that the tactics have a lot of quirky details that are hard to describe.  Hard to know whether the quirks are intentional.  Documenting them essentially makes them part of the spec for Coq.

I had noticed this corner case, which is a bit unrealistic because `reflexivity` will work as long `P1 -> ... -> Pn -> E1 = E2`

```
Goal 1 = 2 -> False -> 1 + 2 = 3.
intro.
reflexivity.
```